src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,165 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Feel free to remove this class and create your own.
+ */
+public class RegistrationServiceTest {
+    private static RegistrationService service;
+    private static StorageDao storageDao;
+    private User alice;
+    private User bob;
+    private User david;
+    private User userWithDuplicatedLogin;
+    private User userWithAgeUnder18;
+    private User userWithSmallPassword;
+    private User userWithNullAge;
+    private User userWithNullLogin;
+    private User userWithNullPassword;
+    private User userWithAllFieldsNull;
+    private User nullUser;
+    private User userWithAgeMoreThan100;
+
+    @BeforeAll
+    static void beforeAll() {
+        service = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        alice = new User("user230j", "de^wdki", 18);
+        bob = new User("user22d0j", "de^wdkHBIUi", 19);
+        david = new User("userDavid", "dewiuydki", 21);
+    }
+
+    @Test
+    void register_validUserCase_Ok() {
+        service.register(alice);
+        service.register(bob);
+        service.register(david);
+
+        int actualSize = Storage.people.size();
+        assertEquals(3, actualSize, "Expected size should be 3, but was: " + actualSize);
+        User actual = storageDao.get(alice.getLogin());
+        assertEquals(alice, actual, "actual should be " + alice + " , but was: " + actual);
+        actual = storageDao.get(bob.getLogin());
+        assertEquals(bob, actual, "actual should be " + bob + " , but was: " + actual);
+        actual = storageDao.get(david.getLogin());
+        assertEquals(david, actual, "actual should be " + david + " , but was: " + actual);
+        Storage.people.clear();
```

Comment:
you have this line in all tests move it to the method with annotation @AfterEach